### PR TITLE
use get instead of list http method as list http method is unsupported in some envs

### LINKED ISSUE #2: Cannot target vaults behind envoy proxy because LIST method is unsupported
When using safe/vaultv and targeting a vault deployed behind envoy proxy (in our case Ambassador), safe paths and safe export do not work.

```
➜  ~/othercode/safe git:(master) ✗ safe paths
!! Could not parse response body as JSON, and returned Content-Type is `text/plain'. Client may not be reaching Vault
````

with  

```
DEBUG=true safe paths 
...
Request:
LIST /v1/secret HTTP/1.1
Host: vault.myk8scluster.com:443


Response:
HTTP/1.1 400 Bad Request
Content-Length: 11
Connection: keep-alive
Content-Type: text/plain
Date: Thu, 20 May 2021 19:56:43 GMT
Server: envoy

Bad Request
!! Could not parse response body as JSON, and returned Content-Type is `text/plain'. Client may not be reaching Vault
```

This is because vaultkv uses the LIST http method, and that isn't supported by envoy 

```
➜  ~/othercode/safe git:(master) ✗ curl  -X LIST -H "X-Vault-Token: REDACTED" https://vault.myk8scluster.com:443/v1/secret
Bad Request%
```
works with GET 
```
➜  ~/othercode/safe git:(master) ✗ curl  -X GET -H "X-Vault-Token: REDACTED" https://vault.myk8scluster.com443/v1/secret\?list\=true
{"request_id":"0669c119-a5bf-8d5f-4f5e-7bac4d06866d","lease_id":"","renewable":false,"lease_duration":0,"data":{"keys":["somekey/"]},"wrap_info":null,"warnings":null,"auth":null}
```

### PR DESCRIPTION
**Pull request by scottillogical**
fixes #2

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,6 +3,9 @@ module github.com/cloudfoundry-community/vaultkv
 go 1.15
 
 require (
-	github.com/onsi/ginkgo v1.13.0
+	github.com/onsi/ginkgo v1.16.2
```
Comment (scottillogical) [inline go.mod:6]: I can revert these go mod changes they are undesirable desirable, I happened to update ginkgo and it appeared to work fine

**Comment by thomasmitchell** [comment]
seems fine - however, I am curious, what is preventing the LIST method from operating in some envs?

**Comment by scottillogical** [comment]
@thomasmitchell envoy proxy is rejecting the request 

```
Request:
LIST /v1/secret HTTP/1.1
Host: vault.myk8scluster.com:443

Response:
HTTP/1.1 400 Bad Request
Content-Length: 11
Connection: keep-alive
Content-Type: text/plain
Date: Thu, 20 May 2021 19:56:43 GMT
Server: envoy

Bad Request
```

Unfortunately couldn't find anything in the envoy docs referencing the supported HTTP methods, but a list is hardcoded here https://github.com/envoyproxy/envoy/blob/a12869fa9e9add4301a700978d5489e6a0cc0526/api/envoy/api/v2/core/base.proto#L42.       You can see someone in the vault issue tracker posting the same issue here https://github.com/hashicorp/vault/issues/9874   I could of course try get envoy to support LIST but seems like using GET is better

**Comment by thomasmitchell** [comment]
yeah, I agree. I'll run this through some tests, but seems straightforward and good. vaultkv exists to increase compatibility - this is definitely in line with that mission :)

**Comment by thomasmitchell** [comment]
thanks for the contribution. I tagged it v0.3.1, so hopefully pkg.go.dev scrapes and pulls it in soon.

**Pull request merged**